_api-reference/document-apis/update-document.md

```diff
@@ -14,6 +14,11 @@ redirect_from:
 If you need to update a document's fields in your index, you can use the update document API operation. You can do so by specifying the new data you want to be in your index or by including a script in your request body, which OpenSearch runs to update the document. By default, the update operation only updates a document that exists in the index. If a document does not exist, the API returns an error. To _upsert_ a document (update the document that exists or index a new one), use the [upsert](#using-the-upsert-operation) operation.
 
 
+You cannot specify any ingest pipeline explicitly when calling the update document API, but the pre-defined default pipeline or final pipeline can be executed under some conditions. Case that when performing upsert operation and a new document is indexed, the pre-defined default pipeline or final pipeline in the index is supposed to be executed. However, when performaing update operation for the existing document, the default pipeline or final pipeline is not supposed to be executed but actually it's executed, a deprecation warning header as follows will be returned, and the support will be removed in 3.0.0.
```

Comment:
```suggestion
You cannot explicitly specify an ingest pipeline when calling the Update Document API. If an ingest pipeline is defined on your index, the following behavior applies:

- **Upsert operations**: When indexing a new document, the ingest pipeline defined in the index is executed.  
- **Update operations**: When updating an existing document, ingest pipeline execution is not recommended because it may produce erroneous results. Support for running ingest pipelines during update operations is deprecated and will be removed in version 3.0.0. If your index has a defined ingest pipeline, the update document operation will return the following deprecation warning: 
```

Reply:
Thanks!

Both `default pipeline` or `final pipeline` can be set in the index settings, like this:
```
PUT test_1
{
  "settings":{
    "index.default_pipeline":"pipeline1",
     "index.final_pipeline":"pipeline2"
  }
}
```
, default pipeline will be executed if there's no pipeline specified when indexing, final pipeline will be executed after the execution of the specified pipeline or default pipeline.

I think we should retain the phrase `default pipeline` and `final pipeline` because for update document API, only these type of pipeline can be executed.

Reply:
@gaobinlong Thank you for the explanation! I edited the suggestion: please take a look and let me know if this captures the intended meaning.

Reply:
It looks good to me, thanks!